File: ruby/src/hello_modules.rb
# frozen_string_literal: true

class HelloController
  def greeting
    "Hello"
  end

  def say(say_to)
    puts "#{greeting}, #{say_to}!"
  end
end

def parse_args
  {"world" => ARGV[0]}
end

def main
  args = parse_args
  controller = HelloController.new
  controller.say args["world"]
end

main

File: elixir/src/hello_modules.exs
defmodule HelloController do
  @greeting "Hello"

  def parse_args(args) do
    %{world: Enum.at(args, 0)}
  end

  def say(to) do
    IO.puts("#{@greeting}, #{to}!")
  end
end

controller = HelloController

_ =
  System.argv()
  |> controller.parse_args
  |> Map.get(:world)
  |> controller.say
